src/System.Private.CoreLib/shared/System/Threading/SemaphoreSlim.cs

```diff
@@ -900,11 +901,11 @@ protected virtual void Dispose(bool disposing)
         private static Action<object> s_cancellationTokenCanceledEventHandler = new Action<object>(CancellationTokenCanceledEventHandler);
         private static void CancellationTokenCanceledEventHandler(object obj)
         {
-            SemaphoreSlim semaphore = obj as SemaphoreSlim;
-            Debug.Assert(semaphore != null, "Expected a SemaphoreSlim");
-            lock (semaphore.m_lockObj)
+            Debug.Assert(obj is Semaphore, "Expected a SemaphoreSlim");
```

Comment:
```suggestion
            Debug.Assert(obj is SemaphoreSlim, "Expected a SemaphoreSlim");
```

Although I wonder why they didn't choose to make it an `Action<SemaphoreSlim>`

Reply:
> Although I wonder why they didn't choose to make it an `Action<SemaphoreSlim>`

It's passed to something that expects an `Action<object>`.

Reply:
> Debug.Assert(obj is SemaphoreSlim

Good catch. Thanks.